tests/network/vmi_istio.go

```diff
@@ -0,0 +1,207 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+const (
+	istioDeployedEnvVariable     = "KUBEVIRT_DEPLOY_ISTIO"
+	istioInjectSidecarAnnotation = "sidecar.istio.io/inject"
+	istioInjectNamespaceLabel    = "istio-injection"
+	vmiDefaultPortsAppSelector   = "istio-vmi-default-ports"
+	vmiExplicitPortsAppSelector  = "istio-vmi-explicit-ports"
+	vmiDefaultPortsAppCaseName   = "vmiWithDefaultPorts"
+	vmiExplicitPortsAppCaseName  = "vmiWithExplicitPorts"
+	svcDeclaredTestPort          = 1500
+	svcUndeclaredTestPort        = 1501
+)
+
+type vmiAppCase struct {
+	appSelector string
+	vmi         *v1.VirtualMachineInstance
+	ports       []v1.Port
+	service     *corev1.Service
+}
+
+func istioServiceMeshDeployed() bool {
+	value := os.Getenv(istioDeployedEnvVariable)
+	if value == "true" {
+		return true
+	}
+	return false
+}
+
+func withAppLabel(value string) libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Labels == nil {
+			vmi.Labels = map[string]string{}
+		}
+		vmi.Labels["app"] = value
+	}
+}
+
+func withIstioAnnotation() libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Annotations == nil {
+			vmi.Annotations = map[string]string{}
+		}
+		vmi.Annotations[istioInjectSidecarAnnotation] = "true"
+	}
+}
+
+func newServiceWithAppSelector(name, namespace, selector string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(appLabel string, network *v1.Network, ports []v1.Port) *v1.VirtualMachineInstance {
+	vmi := libvmi.NewCirros(
+		libvmi.WithNamespace(tests.NamespaceTestDefault),
+		libvmi.WithNetwork(network),
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		withIstioAnnotation(),
+		withAppLabel(appLabel),
+	)
+	tests.AddServiceAccountDisk(vmi, "default")
```

Comment:
Please add this (copied from your doc) or similar comment - " The serviceaccountdisk is required to have the service account token mounted, because it’s used by istio-proxy."

Reply:
Added the comment